ls1intum/Artemis
# `Development`: Add git commit short ID to the startup information box

### Checklist
#### General


#### Server

### Motivation and Context


At the moment (especially for development environments) it is hard to say. which version of Artemis is currently running (or used for CI). 

### Description
As discussed in the deployment working group, we want to add a new information in the startup box, that currently looks like this:
```
----------------------------------------------------------
        'Artemis' is running! Access URLs:
        Local:      http://localhost:8080/
        External:   http://172.19.0.3:8080/
        Profiles:   [dev, bamboo, bitbucket, jira, artemis]
        Version:    5.9.9
        External:   141081
----------------------------------------------------------
```
To uniquely identify the current running version, the best value for that is the current git id, that represents the last commit. 
In order to get this id, we use the `com.gorylenko.gradle-git-properties` as suggested by the spring documentation:
https://docs.spring.io/spring-boot/docs/1.4.1.RELEASE/reference/html/howto-build.html#howto-git-info


### Steps for Testing

1. Build artemis with `./gradlew -Pprod -Pwar clean bootWar`
2. Start artemis and wait for the information box 
3. Compare the displayed git id with the current git id (should be equal)

Alternative you could
1. Look at this log file:
https://bamboo.ase.in.tum.de/download/ARTEMIS-AETG836-DA/build_logs/ARTEMIS-AETG836-DA-2.log
2. Then search for `Git Commit:`.
3. You will see the commit id:
![image](https://user-images.githubusercontent.com/1368405/183925991-a4ec4678-908f-4604-b3d7-a8214c079c2c.png)
4. Which is the same as the commit ID on github:
![image](https://user-images.githubusercontent.com/1368405/183926133-33f13d8a-8c6a-48bd-83dd-59e6d65c541f.png)




### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots
